microshift_storage/volume-snapshots-microshift.adoc

```diff
@@ -0,0 +1,97 @@
+:_content-type: ASSEMBLY
+[id="working-with-volume-snapshots-microshift"]
+= Working with volume snapshots
+include::_attributes/attributes-microshift.adoc[]
+:context: volume-snapshots-microshift
+
+toc::[]
+
+This document describes how to work with volume snapshots to help protect against data loss by using the supported {product-title} logical volume manager storage (LVMS) Container Storage Interface (CSI) provider. Familiarity with xref:../microshift_storage/understanding-persistent-storage-microshift.adoc#persistent-volumes_understanding-persistent-storage-microshift[persistent volumes] is required.
+
+A snapshot represents the state of the storage volume in a cluster at a particular point in time. Volume snapshots can also be used to provision new volumes. Snapshots are created as read-only logical volumes (LVs) located on the same device as the original data.
+
+With CSI volume snapshots, a cluster administrator can:
+
+* Create a snapshot of an existing persistent volume claim (PVC).
+* Backup a volume snapshot to a secure location.
+* Restore a volume snapshot as a different PVC.
+* Delete an existing volume snapshot.
+
+[IMPORTANT]
+====
+Support is only available for the {product-title} CSI driver. In-tree and FlexVolumes are not supported.
```

Comment:
Lots of other CSI implementations are not supported. Why are those 2 called out specifically here?

Reply:
Scooped this note up from the RHEL docs that included it. Wasn't sure of relevance to our use, so left it for review. Remove?

Reply:
Oh, yeah, if this is adapted from something in RHEL (vs. OCP) then I think we can remove it. What do you think @DanielFroehlich ?

Reply:
It is actually in OCP, too. 
FlexVolumes for persistent storage is deprecated. https://docs.openshift.com/container-platform/4.13/storage/persistent_storage/persistent-storage-flexvolume.html 
And there's the exact bullet here: https://docs.openshift.com/container-platform/4.13/storage/container_storage_interface/persistent-storage-csi-snapshots.html#persistent-storage-csi-snapshots-overview_persistent-storage-csi-snapshots 
The "in-tree" occurrences all seem to apply to specific providers (Azure, VMWare, etc.).

Reply:
OK, that's not quite saying the same thing.

> Support is only available for CSI drivers. In-tree and FlexVolumes are not supported.

OCP supports lots of CSI drivers. There are other types of storage access methods that are not supported, and those legacy methods are being listed. 

Neither of those legacy methods ever applied to MicroShift. We will, I'm sure, have other CSI drivers at some point in the future. Let's not make maintenance work for ourselves by being overly specific here. I think you can just remove the note entirely.

Reply:
Flex volumes and "in-tree" volumes are vestigial dynamic provisiong strategies that predate CSI. The logic for both variants is still active in upstream kube so the disclaimer is technically relevant, even if MicroShift users have probably never heard of either.